### PR TITLE
chore(test-data): create `ProductVariant`

### PR DESCRIPTION
**Pull request by jaikamat**
## Summary

This PR creates the `ProductVariant` model and associated draft. Additional comments inline ✨

### CODE REVIEW

File: models/product-variant/src/product-variant-draft/builder.spec.ts [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+        ]),
+        images: expect.arrayContaining([
+          expect.objectContaining({
+            url: expect.any(String),
+            /**
+             * This is not the correct typename, but it saves us from having to create a dummy ImageDraft.
+             *
+             * TODO: pull in ImageDraft in the generator when it's created
+             */
+            __typename: 'Image',
```
Comment (jaikamat) [inline models/product-variant/src/product-variant-draft/builder.spec.ts:83]: @valoriecarli relevant to the `Image` stuff we were talking about 👍

File: models/product-variant/src/transformers.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+      fields: {
+        attributes,
+        isMatchingVariant,
+        scopedPrice,
+        scopedPriceDiscounted,
+        ...rest
+      },
+    }) => ({
+      ...rest,
+      attributesRaw: attributes as Array<TAttributeGraphql>,
```
Comment (jaikamat) [inline models/product-variant/src/transformers.ts:24]: That `core` change makes it much easier to use the prebuilt fields we define in `buildFields`!

File: models/product-variant/src/generator.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+// https://docs.commercetools.com/api/projects/products#productvariant
+
+const generator = Generator<TProductVariant>({
+  fields: {
+    id: fake((f) => f.datatype.uuid()),
+    key: fake((f) => f.lorem.slug(2)),
+    sku: fake((f) => f.random.word()),
+    prices: fake(() => [Price.random()]),
+    price: fake(() => Price.random()),
```
Comment (stephsprinkle) [inline models/product-variant/src/generator.ts:15]: The documentation says

> Only available when [Price selection](https://docs.commercetools.com/api/projects/products#price-selection) is used.

Should this be `null`?

File: models/product-variant/src/generator.ts [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    attributes: fake(() => [Attribute.random()]),
+    images: fake(() => [Image.random()]),
+    assets: [],
+    availability: null,
+    isMatchingVariant: fake((f) => f.datatype.boolean()),
+    scopedPrice: null,
+    scopedPriceDiscounted: fake((f) => f.datatype.boolean()),
```
Comment (stephsprinkle) [inline models/product-variant/src/generator.ts:22]: Can this also be `null` since `scopedPrice` is null?

**Review by stephsprinkle — APPROVED** [review]

File: models/product-variant/src/transformers.ts (hunk 2 above)
Comment (emmenko) [reply models/product-variant/src/transformers.ts:24]: Nice!

File: models/product-variant/src/generator.ts [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+    id: fake((f) => f.datatype.uuid()),
+    key: fake((f) => f.lorem.slug(2)),
+    sku: fake((f) => f.random.word()),
+    prices: fake(() => [Price.random()]),
+    price: null,
+    attributes: fake(() => [Attribute.random()]),
+    images: fake(() => [Image.random()]),
+    assets: [],
+    availability: null,
+    isMatchingVariant: fake((f) => f.datatype.boolean()),
```
Comment (emmenko) [inline models/product-variant/src/generator.ts:20]: I would keep it as `false` by default. This is only used for product projection search results.

image

**Pull request merged**